build.gradle

```diff
@@ -101,6 +92,22 @@ dependencies {
 
     implementation configurations.agentDeps
 
+    implementation ('commons-beanutils:commons-beanutils:1.9.4') {
```

Comment:
Would be good to have a discussion on this whole snyk lark for scanning gradle files.  I was astounded when we first plumbed it in (using scanning of the build.gradle by the website) that it came up with so many false positives.  It looks like this is still the case, even with the gradle file scanning - unless hopefully I'm missing something?
You have added 5 implementation dependencies here, which means that they will be included in the final community api jar file.  However only 2 of the dependencies (dom4j and snakey) are actually implementation dependencies anyway so the other 3 aren't required and shouldn't be added.  This can be confirmed by running 
```
./gradlew clean assemble && jar -tvf build/libs/community-api-2020-04-28.jar| grep -e beanutils -e collection -e dom4j -e xerces -e snakey
```
on both master and then on this branch.

Reply:
this is interesting and seems to be a limitation of the tool and the way it implements the scan. 

however the output of the tool is very informative, and looking at the output for this particular project, all the vulnerabilities which do not appear in the final jar are introduced through `org.codehaus.groovy.modules.http-builder`, which i can see is only a `testImplementation` dependency. the best option is probably just to ignore these in the `.snyk` file.  i'll make the change and update the PR. 

i suppose scenarios like this are good examples of why the org as a whole needs to have some best practices defined for responding to the output of these types of tools.

Reply:
FYI here is the output i'm referencing above

```
Tested 193 dependencies for known issues, found 7 issues, 16 vulnerable paths.


Issues with no direct upgrade or patch:
  ✗ Arbitrary Code Execution [High Severity][https://snyk.io/vuln/SNYK-JAVA-COMMONSBEANUTILS-30077] in commons-beanutils:commons-beanutils@1.8.0
    introduced by org.codehaus.groovy.modules.http-builder:http-builder@0.7.1 > net.sf.json-lib:json-lib@2.3 > commons-beanutils:commons-beanutils@1.8.0
  This issue was fixed in versions: 1.9.2
  ✗ Deserialization of Untrusted Data [High Severity][https://snyk.io/vuln/SNYK-JAVA-COMMONSCOLLECTIONS-30078] in commons-collections:commons-collections@3.2.1
    introduced by org.codehaus.groovy.modules.http-builder:http-builder@0.7.1 > net.sf.json-lib:json-lib@2.3 > commons-collections:commons-collections@3.2.1
  This issue was fixed in versions: 3.2.2
  ✗ Deserialization of Untrusted Data [Medium Severity][https://snyk.io/vuln/SNYK-JAVA-COMMONSCOLLECTIONS-472711] in commons-collections:commons-collections@3.2.1
    introduced by org.codehaus.groovy.modules.http-builder:http-builder@0.7.1 > net.sf.json-lib:json-lib@2.3 > commons-collections:commons-collections@3.2.1
  This issue was fixed in versions: 3.2.2
  ✗ XML External Entity (XXE) Injection [High Severity][https://snyk.io/vuln/SNYK-JAVA-ORGDOM4J-565810] in org.dom4j:dom4j@2.1.1
    introduced by org.hibernate:hibernate-java8@5.4.5.Final > org.hibernate:hibernate-core@5.4.12.Final > org.dom4j:dom4j@2.1.1 and 1 other path(s)
  This issue was fixed in versions: 2.1.3, 2.0.3
  ✗ Denial of Service (DoS) [Medium Severity][https://snyk.io/vuln/SNYK-JAVA-ORGYAML-537645] in org.yaml:snakeyaml@1.25
    introduced by org.springframework.boot:spring-boot-starter-actuator@2.2.6.RELEASE > org.springframework.boot:spring-boot-starter@2.2.6.RELEASE > org.yaml:snakeyaml@1.25 and 8 other path(s)
  This issue was fixed in versions: 1.26
  ✗ Denial of Service (DoS) [Medium Severity][https://snyk.io/vuln/SNYK-JAVA-XERCES-30183] in xerces:xercesImpl@2.11.0
    introduced by org.codehaus.groovy.modules.http-builder:http-builder@0.7.1 > net.sourceforge.nekohtml:nekohtml@1.9.22 > xerces:xercesImpl@2.11.0
  This issue was fixed in versions: 2.12.0
  ✗ Denial of Service (DoS) [High Severity][https://snyk.io/vuln/SNYK-JAVA-XERCES-31585] in xerces:xercesImpl@2.11.0
    introduced by org.codehaus.groovy.modules.http-builder:http-builder@0.7.1 > net.sourceforge.nekohtml:nekohtml@1.9.22 > xerces:xercesImpl@2.11.0
  This issue was fixed in versions: 2.12.0
```

Reply:
on closer inspection it seems that `http-builder` isn't even used in the tests, so i've removed it and removed the snyk policy entries too.